doc/tutorials/nvmx_virtualization.rst

```diff
@@ -0,0 +1,315 @@
+   cp kernel_config_uefi_sos .config
+   make olddefconfig
+
+The following configuration entries are needed to launch nested
+guests on the Service VM:
+
+.. code-block:: none
+
+   CONFIG_KVM=y
+   CONFIG_KVM_INTEL=y
+   CONFIG_ACRN_GUEST=y
+
+This setting is also needed if you want to use virtio block device as root filesystem
+for the guest VMs:
+
+.. code-block:: none
+
+   CONFIG_VIRTIO_BLK=y
+
+After the configuration modifications, build and install the kernel.
+
+.. code-block:: none
+
+   make all
+
+
+Launch a Nested Guest VM
+************************
+
+Create an Ubuntu KVM Image
+==========================
+
+Refer to :ref:`Build the Ubuntu KVM Image <build-the-ubuntu-kvm-image>`
+on how to create an Ubuntu KVM image as the nested guest VM's root filesystem.
+There is no particular requirement for this image, e.g., it could be of either
+qcow2 or raw format.
+
+Prepare for Launch Scripts
+==========================
+
+Install QEMU on the Service VM that will launch the nested guest VM:
+
+.. code-block:: none
+
+   sudo apt-get install qemu-kvm qemu virt-manager virt-viewer libvirt-bin
+
+The following is a simple example for the script to launch a nested guest VM.
+
+.. important:: The ``-cpu host`` option is needed to launch a nested guest VM, and ``-nographics``
+   is required to run nested guest VMs reliably.
+
+Besides this, there is no particular requirements for the launch script.
+You can prepare the script just like the one you use to launch a VM
+on native Linux.
+
+For example, other than ``-hda``, you can use the following option to launch
+a virtio block based RAW image: ``-drive format=raw,file=/root/ubuntu-20.04.img,if=virtio``
+
+Use the following option to enable Ethernet on the guest VM:
+``-netdev tap,id=net0 -device virtio-net-pci,netdev=net0,mac=a6:cd:47:5f:20:dc``
+
+.. code-block:: bash
+   :emphasize-lines: 2-4
+
+   sudo qemu-system-x86_64 \
+     -enable-kvm \
+     -cpu host \
+     -nographic \
+     -m 2G -smp 2 -hda /root/ubuntu-20.04.qcow2 \
+     -net nic,macaddr=00:16:3d:60:0a:80 -net tap,script=/etc/qemu-ifup
+
+Launch the Guest VM
+===================
+
+You can launch the nested guest VM from the Service VM's virtual serial console
+or from an SSH remote login.
+
+If the nested VM is launched successfully, you should see the nested
+VM's login prompt:
+
+.. code-block:: bash
+
+   [  OK  ] Started Terminate Plymouth Boot Screen.
+   [  OK  ] Started Hold until boot process finishes up.
+   [  OK  ]  Starting Set console scheme...
+   [  OK  ] Started Serial Getty on ttyS0.
+   [  OK  ] Started LXD - container startup/shutdown.
+   [  OK  ] Started Set console scheme.
+   [  OK  ] Started Getty on tty1.
+   [  OK  ] Reached target Login Prompts.
+   [  OK  ] Reached target Multi-User System.
+   [  OK  ] Started Update UTMP about System Runlevel Changes.
+
+   Ubuntu 20.04 LTS ubuntu_vm ttyS0
+
+   ubuntu_vm login:
+
+You won't see the nested guest from a ``vcpu_list`` or ``vm_list`` command
```

Comment:
@NanlinXie @junjiemao1 Can the ``vcpu_list`` and ``vm_list`` commands be updated to show these nested VMs?

Reply:
> @NanlinXie @junjiemao1 Can the `vcpu_list` and `vm_list` commands be updated to show these nested VMs?

No, nested VMs are managed by L1 hypervisor not ACRN, so  HV commands can't show them.

Reply:
Is there an L1 hypervisor (KVM) command to show these nested VMs?

Reply:
AFAIK there is no such command available in Linux to show QEMU launched guests.

Reply:
I'm not aware of any other mean of listing these QEMU VMs short of grepping for all `qemu*` processes running on the host.

What we could do as an enhancement to this guide is to create the VM and start/stop it using `virsh`. That way you can also list them all.

Reply:
Yes, that'a true. I think this guide is focusing on ACRN and the 'nested' specific features.In terms of how to launch guest from SOS, it's not important, and users can learn it from generic QEMU/libvirt documentations.